tools/c7n_tencentcloud/c7n_tencentcloud/resources/cos.py

```diff
@@ -0,0 +1,355 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+
+import jmespath
+from qcloud_cos import CosS3Client, CosConfig, CosServiceError
+
+from c7n.filters import Filter, ValueFilter
+from c7n_tencentcloud.provider import resources
+from c7n_tencentcloud.query import QueryResourceManager, ResourceTypeInfo, DescribeSource
+from c7n.utils import type_schema, format_string_values
+from tencentcloud.common import credential
+
+log = logging.getLogger('custodian.cos')
+
+
+class DescribeCos(DescribeSource):
+
+    @staticmethod
+    def get_cos_client(region):
+        cred = credential.DefaultCredentialProvider().get_credentials()
```

Comment:
Why not just use the session factory and utilize the credentials from there? this doesnt handle assume role for example

Reply:
done, credentials using the session_factory